scripts/kconfig/kconfigfunctions.py

```diff
@@ -0,0 +1,62 @@
+#
+# Copyright (c) 2018-2019 Linaro
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import sys
+import os
+import re
+
+# Types we support
+# 'string', 'int', 'hex', 'bool'
+conf_file = os.path.join(os.environ['PROJECT_BINARY_DIR'], "include/generated/generated_dts_board.conf")
+
+dt_defines = {}
+if os.path.isfile(conf_file):
+    for line in open(conf_file, 'r', encoding='utf-8'):
+        if re.search('=', line):
+            c = line.split('=')
+            dt_defines[c[0]] = c[1].strip()
+
+def _dt_units_to_scale(unit):
+    if not unit:
+        return 0
+    if unit == 'k' or unit == 'K':
+        return 10
+    if unit == 'm' or unit == 'M':
+        return 20
+    if unit == 'g' or unit == 'g':
+        return 30
+
+def dt_get_int_val(kconf, name, arg, unit=None):
+    try:
+        d = dt_defines[arg]
+        if d.startswith('0x') or d.startswith('0x'):
+            d = int(d, 16)
+        else:
+            d = int(d)
+        d = d >> _dt_units_to_scale(unit)
+
+    except:
+        d = 0
+    return str(d)
+
+def dt_get_hex_val(kconf, name, arg, unit=None):
+    try:
+        d = dt_defines[arg]
+        if d.startswith('0x') or d.startswith('0x'):
+            d = int(d, 16)
+        else:
+            d = int(d)
+        d = d >> _dt_units_to_scale(unit)
+        d = hex(d)
+
+    except:
```

Comment:
It is not OK to catch "all" exceptions, as that can lead to impossibly difficult debugging when an unexpected exception occurs, like OOM / syntax error, etc.